### ISSUE TITLE
Next release version

### ISSUE THREAD

**Issue by mterzo** [voxpupuli/puppetboard#426]
Should we finally make the leap to 1.0.0 for the next release?

**Comment by kirkins** [comment]
Makes sense to me, given #421

**Comment by larsnaesbye** [comment]
Sounds reasonable to me too. Anything in particular we'd like to mark as milestones for v1.0 ?

**Comment by tux-o-matic** [comment]
Wasn't the original goal of the project to offer a PuppetDB backed dashboard to replace the old Ruby dashboard?
I think that goal was met.

**Comment by larsnaesbye** [comment]
I agree. It is way better even. 
The only feature I miss from the original Puppet Dashboard is the CSV export, but I know someone who's looking into this :)

**Comment by larsnaesbye** [comment]
My suggested milestone list for 1.0.0:

- #30  (related to #428  )
- #283
- #333
- #360 
- #361 
- an update to jQuery 3.x and Semantic UI 2.x
- new screenshots

**Comment by tux-o-matic** [comment]
#283 and #360 are pretty much the same.

And a general reminder, if someone would like to remove Python 2.6 support from release 1.0.0, remember that 2.6 is still the Python shipped on EL 6. Since containers can't run properly on the EL 6 Kernel, only alternative would be to install a newer Python from Software Collections.

**Comment by larsnaesbye** [comment]
I picked those two because they were already planned for 0.3.0 at the time.

**Comment by emlun007** [comment]
I suggest using new script for sorting tables, as current sorter cannot handle ip adresses, time (days, hours) and memory

**Comment by larsnaesbye** [comment]
Any suggestions for a better sorter, @emlun007 ? :)

**Comment by emlun007** [comment]
@larsnaesbye , https://github.com/christianbach/tablesorter 
Should be able to sort ip and time

**Comment by larsnaesbye** [comment]
Or maybe we should simply make a 1.0 release now?

**Comment by larsnaesbye** [comment]
@mterzo - any things you think holds a 1.0 release back?

**Comment by kirkins** [comment]
@larsnaesbye why do you care so much. It's just a number.

Also there are so many issues open right now. Fix something instead of bothering about version number over and over again.

I don't mean to be rude but I honestly feel like @mterzo is getting spammed lately.

Also @emlun007 if you open a PR it should be working or at least semi-working, if it is semi-working describe what works and what doesn't.

**Comment by emlun007** [comment]
@kirkins, PR #429 (button for saving csv) works. I have been using the function for 2 months now. I am just waiting for code to be merged.

**Comment by kirkins** [comment]
@emlun007 you have a point, I have had PR open for a long time too.

I'm frustrated with this repo because I get so notifications but there are never any updates.

Someone else should be added to the project to sort through everything...

**Comment by kirkins** [comment]
@larsnaesbye sorry if that was harsh. I'm having a rough morning. Cheers

**Comment by emlun007** [comment]
@kirkins, I have code for new PRs but I don't want to spam too much, so I'm just waiting for the first one to be merged. It's been three months soon 🐌 🐌 🐌

**Comment by larsnaesbye** [comment]
@kirkins no problem mate. I'll go fix one of the things alright :-)
I was trying to figure out a kind of priority list before we can re-pull some of the good changes already in. It came out wrong.

**Comment by florinbesleaga** [comment]
Hi all. Is it possible, by any chance, to RELEASE THE KRAKEN ?
I am only able to 'puppet install ..' , so a new release would be extremely helpful.

**Comment by larsnaesbye** [comment]
@florinbesleaga Now the codename **has** to be 1.0 Kraken :-D

**Comment by larsnaesbye** [comment]
@wyardley Thank you very much for the pull request merge. Do you think it's possible to make a release from the current trunk, as version 0.3.4? It would help some of us quite a bit.

**Comment by othalla** [comment]
Agreed! Would be great!

**Comment by wyardley** [comment]
@larsnaesbye I haven’t worked on this project much so probably someone else should cut the release. Maybe post in the Vox channel on IRC and see if someone can do that.

**Comment by blackknight36** [comment]
Is this project even active?

**Comment by larsnaesbye** [comment]
Somewhat, yes.

**Comment by larsnaesbye** [comment]
I have now joined the Freenode IRC channel #voxpupuli - here's hoping we can find some guidance there :-)

**Comment by larsnaesbye** [comment]
I have had a single conversation on #voxpupuli - but not much have come out of it. There are many good patches waiting for merging, and a potential for a real good new release - but there is no automation set up for the release process, and that kept the good folks over at #voxpupuli from jumping in.

**Comment by blackknight36** [comment]
@larsnaesbye What do they mean by automated release process?  Github has TravisCI available but I'm not familiar with writing tests for python.  It sounds like somebody should just make a 0.3.2 release which contains the patches that have PRs open now.

**Comment by wyardley** [comment]
@mterzo @corey-hammerton: are either of you able to release this and / or grant access to some folks from Vox? I think daenney can be removed. Those are the only 3 people with publish permission to the PyPI currently.

@blackknight36 ideally there would be some sort of option to publish via CI as well, similar to what we use with Puppet modules, but I don't think Vox really has much tooling around publishing Python packages.

**Comment by larsnaesbye** [comment]
@blackknight36 I got the message that there wasn't set up an easily triggerable process for making a release on this project.

**Comment by anarcat** [comment]
having any new release whatsoever would be a good start. ;)

**Comment by larsnaesbye** [comment]
Pushing the issue on #voxpupuli IRC channel like I did might help. It did make some waves...

**Comment by larsnaesbye** [comment]
@bastelfreak , could you perhaps tell us what to do? I'm sure many of us would like to help, but the steps to a new release are somewhat unclear....

**Comment by ZeroPointEnergy** [comment]
@mterzo It would be really great if someone with the rights to do so could enable the people who try to help here to actually move the project forward. I think there is mainly an issue around the release process? Every other month something here flairs up and there is once again hope it will be revived.

There is clearly the need of an open source dashboard that uses the awesome power of the PuppetDB. In a company I work for we currently use Puppetboard as it is the only one available with an acceptable feature set, but it has some serious scaleability issues and there are other features we would like to implement. It is clear that we need a solution and at the moment we are asking ourselves if it is even worth to attempt to fix it if the project is abandoned anyway or if we should just start over with yet another dashboard project.

**Comment by anarcat** [comment]
@ZeroPointEnergy if you have a company using this free software project that is avaialble for free, maybe you should consider getting involved in funding its development instead of dropping it for another one, or starting your own (see also [this article](https://blog.tidelift.com/open-source-has-a-working-for-free-problem) for context).

that said, i am not sure the problem is enabling people as much as just sitting down and just doing the work. this is a Python project, so I think doing a release probably involves simply:

 1. lay a tag
 2. upload the release to PyPI

From what i can tell, the people that can do the former are @mterzo and anyone in the @voxpupuli team (which is quite a lot of people, possibly). The latter is similar, the [PyPI maintainers](https://pypi.org/project/puppetboard/) include @mterzo @corey-hammerton  @daenney and, again, all of @voxpupuli.

Obviously, the problem is we need someone to step up and actually own up to manage this project, which is harder to do than just drop comments on github issues, unfortunately. :p

**Comment by wyardley** [comment]
@anarcat The last time I had looked, I don't think voxpupuli was a contributer on the PyPI project. But also, I don't think "all" of voxpupuli directly has access -- I'm not sure who has access to the voxpupuliorg account in PyPI, but maybe someone else knows. I am pretty confident that this is not an account that any vox contributor has access to.

**Comment by daenney** [comment]
Look again :wink: 
 
![2019-05-23_21-12](https://user-images.githubusercontent.com/569574/58279761-c37ba880-7d8e-11e9-8ee9-b09bcab86172.png)

The Voxpupuli is probably owned by the PMC, so at the very least they can add additional people there if that's what's holding this back.

If someone wants to build a milestone for 1.0, spawn the necessary tickets, changelog updates etc. I'm happy to do the actual uploading to PyPi.

**Comment by wyardley** [comment]
Yes, I see that it's there now, I just meant that I didn't remember it being there when I looked a while back. Could be wrong though.

**Comment by larsnaesbye** [comment]
@daenney , I guess there is a consensus that 1.0 is pretty much ready (?), so only the changelog and perhaps updated screenshots are needed.

**Comment by ZeroPointEnergy** [comment]
@anarcat I'm not sure why you got the impression I ask someone to work for free?

What I actually said is that the project looks abandoned and therefor it would be hard to justify to get time to work on it. I had the impression from previous comments that voxpupuli can't release to pypi and that is the actual blocker.

**Comment by anarcat** [comment]
On 2019-05-23 22:58:28, Andreas Zuber wrote:
> @anarcat I'm not sure why you got the impression I ask someone to work for free?

Are you offering to pay?

**Comment by daenney** [comment]
Hi people. Can we stop the snapping at each other please? I'd like to get back to the topic on hand, releasing a new version.

**Comment by anarcat** [comment]
> Hi people. Can we stop the snapping at each other please? I'd like to get back to the topic on hand, releasing a new version.

Sure, my apologies.

**Comment by larsnaesbye** [comment]
Nope, they didn’t And it certainly can wait until after a release :)

**Comment by anarcat** [comment]
by the way, i must say i truly appreciate that an actual unicorn came in here to calm me down. thanks @daenney :)

**Comment by larsnaesbye** [comment]
#502 is my attempt at a pull request for a simple changelog and version update.

**Comment by kirkins** [comment]
Great to see you'll release a new version.

I have some open PRs that could be merged in before 1.0.

https://github.com/voxpupuli/puppetboard/pull/441
https://github.com/voxpupuli/puppetboard/pull/438
https://github.com/voxpupuli/puppetboard/pull/439

I'm available if these require any additional changes.

**Comment by volans-** [comment]
Glad to see progress on this!

Me too I have #477 that I'd like to be included if possible. I'll take care of resolving the rebase conflicts tomorrow (sorry, don't have access right now)

**Comment by larsnaesbye** [comment]
I'll adjust the Changelog if more PRs get merged, of course.
But I'm thinking that we shouldn't wait too long with this already delayed release, so maybe we could ask whoever makes the 1.0.0 tag to prepare a 1.0.1 tag too for a release not long after?

**Comment by larsnaesbye** [comment]
@alexjfisher : Sorry to ask you yet again, but do think the PRs #438 #439 #441 and #477 could get merged?

**Comment by ZeroPointEnergy** [comment]
Sorry for commenting and maybe trigger some people again, but why not just give @larsnaesbye commit and publish rights?

**Comment by daenney** [comment]
That's not what we're here to discuss right now and there's a process to go through to do that. Lets stay on topic please. Feel free to file a separate issue and tag the PMC on it.

**Comment by anarcat** [comment]
i agree that we shouldn't get lost in "what possible pending PR we could merge before release problem". if the PR isn't merged, it's not going to get merged before the release. we can always make new releases later. 

if you want your PR to get merged, get it reviewed: that can be done by someone else on your team or any other user of GitHub. right now, there is *zero* PR that has a positive review *and* that passes test (apart from the changelog one), so it's unlikely anyone here will have time to fix that before the release.

let's just get this show on the road!

**Comment by daenney** [comment]
It looks like I caused some confusion there. My response was to they "why don't we just give X commit access".

W.r.t the pending PRs, if people can commit time in the next week to get those reviewed and rebased I think we can wait for that. But if it's going to take more time this risks stalling again which would be a shame. I'm entirely happy to accept external/first-time reviewers on any of the pending PRs, none look particularly complex.

**Comment by kirkins** [comment]
@anarcat one of my PRs is fixing the graph on the node page.

It wasn't merged because someone was going to do it a different way. That was 2 years ago.

I guess I'll come back in another 2 years and see if he had the time.

**Comment by daenney** [comment]
> I guess I'll come back in another 2 years and see if he had the time.

Snark isn't helping. I know it's annoying that PRs linger and for that I'm sorry. However, people have lives too and that can take priority over things. You can use the mailing lists or reach out to the PMC directly if you'd like to bring attention to this problem.

**Comment by kirkins** [comment]
@anarcat as for your rant on not getting paid for this you had above.

I wouldn't pay you a dime. This project had potential and it's been completely neglected. On top of that you haven't even contributed to this project.

For you to come here are start talking about money is an insult to those of us who have donated our time to fixing bugs but were completely ignored.

**Comment by daenney** [comment]
Oy. I've asked people to stop snapping at each other. So stop. I'm not going to ask again. Whatever issues you might have with each other, this is not the place. @anarcat was kind enough to recognize that and leave it for what it was. I expect you to do the same @kirkins.

**Comment by wyardley** [comment]
One other question — is it intentional or accidental that Vox contributors don’t have approval or merge access on this repo anymore? I know there was some team shuffling so maybe accidental?

**Comment by wyardley** [comment]
Also, totally agree with shipping first, then merging more changes vs vice-versa.

**Comment by daenney** [comment]
Accidental-intentional. 

B/c pypuppetdb and puppetboard are very different beasts from most of the other things:
* Not Ruby
* Not a Puppet module

We early on created a separate team to gate access to these repos. Unfortunately the two maintainers appear to have lost interest or be otherwise occupied which caused this to block.

**Comment by alexjfisher** [comment]
@daenney Hi!

> voxpupuliorg is probably owned by the PMC, so at the very least they can add additional people there if that's what's holding this back.

That's indeed correct.  https://github.com/voxpupuli/puppetboard/pull/498#issuecomment-492152720
Looks like someone came through and added the accounts.

**Comment by daenney** [comment]
Yeah they did, thanks @alexjfisher

**Comment by bastelfreak** [comment]
Hi people. I had a discussion with @mterzo a few days ago. We added our PyPi account voxpupuliorg to both projects, so we can now do releases as an org and not only from individual accounts. Both projects have a PR open to automate the release process with travis. Anyone that feels confident with python can review them:
* https://github.com/voxpupuli/pypuppetdb/pull/147
* https://github.com/voxpupuli/puppetboard/pull/503

Let us know if one of you wants to spend more time on one/both projects and we can grant merge permissions here.
